File: tablet/Resources/menu/SetMenu_FROM_LP.js
// SetMenu_FROM_LP.js

/*
    Copyright (C) 2012-2015 Rudolf Cardinal (rudolf@pobox.com).
    Department of Psychiatry, University of Cambridge.
    Funded by the Wellcome Trust.

    This file is part of CamCOPS.

    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at

        http://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
*/

/*jslint node: true, newcap: true */
"use strict";
/*global L */

module.exports = function SetMenu_FROM_LP() {

    var MenuWindow = require('menulib/MenuWindow'),
        UICONSTANTS = require('common/UICONSTANTS'),
        ALLTASKS = require('common/ALLTASKS'),
        data = [
            UICONSTANTS.CHANGE_PATIENT_MENU_LINE,
            // ================================================================
            // GENERIC
            // ================================================================
            {
                maintitle: L('t_from_lp_generic'),
                labelOnly: true
            },
            ALLTASKS.TASKLIST.CGI_I,
            // CORE-10 -- copyright conditions prohibit
            ALLTASKS.TASKLIST.IRAC,
            ALLTASKS.TASKLIST.PATIENT_SATISFACTION,
            ALLTASKS.TASKLIST.FFT,
            ALLTASKS.TASKLIST.REFERRER_SATISFACTION_GEN,
            ALLTASKS.TASKLIST.REFERRER_SATISFACTION_SPEC,
            // ================================================================
            // SPECIFIC
            // ================================================================
            {
                maintitle: L('t_from_lp_specific'),
                labelOnly: true
            },
            ALLTASKS.TASKLIST.ACE3,
            ALLTASKS.TASKLIST.PHQ9,
            // EPDS -- Royal College not currently permitting use
            ALLTASKS.TASKLIST.GAD7,
            ALLTASKS.TASKLIST.HONOS,
            ALLTASKS.TASKLIST.AUDIT_C,
            ALLTASKS.TASKLIST.BMI
            // *** EQ-5D-5L, if permitted?
        ],
        self = new MenuWindow({
            title: L('t_set_from_lp'),
            subtitle: L('s_set_from_lp'),
            data: data
        });

    return self;
};

File: tablet_qt/menu/setmenufromlp.cpp
/*
    Copyright (C) 2012, University of Cambridge, Department of Psychiatry.
    Created by Rudolf Cardinal (rnc1001@cam.ac.uk).

    This file is part of CamCOPS.

    CamCOPS is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    CamCOPS is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with CamCOPS. If not, see <https://www.gnu.org/licenses/>.
*/

#include "setmenufromlp.h"

#include "common/uiconst.h"
#include "lib/uifunc.h"
#include "menulib/menuitem.h"
#include "tasks/ace3.h"
#include "tasks/auditc.h"
#include "tasks/bmi.h"
#include "tasks/cgii.h"
#include "tasks/core10.h"
#include "tasks/epds.h"
#include "tasks/eq5d5l.h"
#include "tasks/fft.h"
#include "tasks/gad7.h"
#include "tasks/honos.h"
#include "tasks/irac.h"
#include "tasks/patientsatisfaction.h"
#include "tasks/phq9.h"
#include "tasks/referrersatisfactiongen.h"
#include "tasks/referrersatisfactionspec.h"

SetMenuFromLp::SetMenuFromLp(CamcopsApp& app) :
    MenuWindow(app, uifunc::iconFilename(uiconst::ICON_SETS_CLINICAL))
{
}

QString SetMenuFromLp::title() const
{
    return "FROM-LP";
}

QString SetMenuFromLp::subtitle() const
{
    return tr(
        "RCPsych Framework for Routine Outcome Measurement in "
        "Liaison Psychiatry (FROM-LP)"
    );
}

void SetMenuFromLp::makeItems()
{
    m_items = {
        MAKE_CHANGE_PATIENT(m_app),

        MenuItem(tr("GENERIC SCALES")).setLabelOnly(),
        MAKE_TASK_MENU_ITEM(CgiI::CGI_I_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Core10::CORE10_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Fft::FFT_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Irac::IRAC_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(PatientSatisfaction::PT_SATIS_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(
            ReferrerSatisfactionGen::REF_SATIS_GEN_TABLENAME, m_app
        ),
        MAKE_TASK_MENU_ITEM(
            ReferrerSatisfactionSpec::REF_SATIS_SPEC_TABLENAME, m_app
        ),

        MenuItem(tr("DISEASE-SPECIFIC SCALES")).setLabelOnly(),
        MAKE_TASK_MENU_ITEM(Ace3::ACE3_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(AuditC::AUDITC_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Bmi::BMI_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Epds::EPDS_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Gad7::GAD7_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Honos::HONOS_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Phq9::PHQ9_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Eq5d5l::EQ5D5L_TABLENAME, m_app),
    };
}
